From bb2fde5ddbc18a2e7795ca4d24759230c2aae9d0 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 3 Aug 2022 01:10:01 +0200
Subject: [klibc] fcntl: Fix build failure for some architectures with Linux
 5.19

Starting from Linux 5.19, the kernel UAPI headers now only define
__ARCH_FLOCK64_PAD if the architecture actually needs padding in
struct flock64.  Wrap its use with #ifdef,

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 usr/include/fcntl.h | 2 ++
 1 file changed, 2 insertions(+)

(limited to 'usr/include/fcntl.h')

diff --git a/usr/include/fcntl.h b/usr/include/fcntl.h
index ed703a62d6c1b..cb2e4e53b3330 100644
--- a/usr/include/fcntl.h
+++ b/usr/include/fcntl.h
@@ -33,7 +33,9 @@ struct flock {
 	__kernel_loff_t l_start;
 	__kernel_loff_t l_len;
 	__kernel_pid_t  l_pid;
+#ifdef __ARCH_FLOCK64_PAD
         __ARCH_FLOCK64_PAD
+#endif
 };
 
 #ifdef F_GETLK64
-- 
cgit 

